src/pratt2.rs

```diff
@@ -0,0 +1,296 @@
+#![allow(missing_docs)]
+
+use super::*;
+
+trait Operator<'a, I, O, E>
+where
+    I: Input<'a>,
+    E: ParserExtra<'a, I>,
+{
+    type Op;
+    type OpParser: Parser<'a, I, Self::Op, E>;
+    const INFIX: bool = false;
+    const PREFIX: bool = false;
+
+    fn op_parser(&self) -> &Self::OpParser;
+    fn associativity(&self) -> Associativity;
+    fn fold_infix(&self, lhs: O, op: Self::Op, rhs: O) -> O {
+        unreachable!()
+    }
+    fn fold_prefix(&self, op: Self::Op, rhs: O) -> O {
+        unreachable!()
+    }
```

Comment:
You can ignore this. Took me a bit to realize that it was so that you could use one trait for all operator types.

Reply:
Yeah, it's a bit weird for sure. In practice, it never gets called since which function gets invoked depends on the compile-time associated constants (like `INFIX`). So in theory, the compiler should totally squash all of this and it never ends up in the final binary. It's a fun hack, I guess!